### DISCUSSION TITLE
Is there a way to get markup to display referenced issue name?

### DISCUSSION THREAD

**Discussion by jdx-john** [community/community#22603] [New to GitHub]
<p>I can easily reference other issues, even in different repos using #.</p>
<p>But while it shows the issue name in a tooltip or in autocomplete while typing, adn the _referenced_ issue gains a link to the new issue-name, my new issue only displays the issue ID.</p>
<p>Is there no way to have it show the issue title which is far more human-readable?</p>

**Comment by nadiajoyce** [comment, marked as answer]
<p>Hi <a class="mention" href="/jdx-john">@jdx-john</a>,</p>
<p>There’s no built in way to do that on our end, but I will share this with the appropriate team as a feature request for future consideration. That said,  you could try using <a href="https://github.com/benbalter/copy-issue-link-bookmarklet" rel="nofollow noopener">this bookmarklet</a> to copy issue titles &amp; URLs in markdown format as a workaround.</p>
<p>Cheers!</p>

**Comment by jdx-john** [comment]
<p>Thanks on both counts :slight_smile:</p>

**Comment by jdx-john** [comment]
<p>To follow this up, I’ve been using that bookmarklet but want to tweak it slightly and lack to web-dev skills to understand the DOM manipulation enough.</p>
<p>The code is </p>
<pre><code>javascript&amp;colon;(function()%7Bvar a,b,c%3Ba%3Ddocument.createElement("textarea"),b%3Ddocument.getElementsByTagName("h1")%5B1%5D.children%5B0%5D.textContent.trim(),c%3Dwindow.location.href,a.value%3D"%5B"%2Bb%2B"%5D("%2Bc%2B")",document.body.appendChild(a),a.select(),document.execCommand("copy"),a.remove()%7D).call(this)%3B
</code></pre>
<p>So if I navigate to <a href="https://github.com/github/hub/issues/1913" rel="nofollow noopener">https://github.com/github/hub/issues/1913</a> and run it I get:</p>
<p>[ability to list reviewers with the --format flag](<a href="https://github.com/github/hub/issues/1913" target="_blank" rel="nofollow noopener">https://github.com/github/hub/issues/1913</a>)</p>
<p>I would like it to include the issue number in the title. e.g.</p>
<p>[<strong>193 -</strong> ability to list reviewers with the --format flag](<a href="https://github.com/github/hub/issues/1913" target="_blank" rel="nofollow noopener">https://github.com/github/hub/issues/1913</a>)</p>
<p>I can see that the code is getting the H1 element and that does contain the number, but in a second &lt;span&gt; element. Equally, it is encoded in the URL but that might get messy to detect if I follow a comment link as it isn’t always the last part of the URL.</p>
<p>For someone who knows this stuff it must be trivial, for me it could take an hour or more mucking about!</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/jdx-john">@jdx-john</a>,</p>
<p>Unfortunately, the bookmarklet code seems to have changed all special characters back into their HTML code and I’m having difficulty converting that back into its proper format. You wouldn’t happen to have a Gist of that code, would you? I’d love to poke at it and see if I can help you make this modification.</p>
<p>Thanks!</p>

**Comment by jdx-john** [comment]
<p>I just got it from the link a couple of posts above…</p>

**Comment by ThirdEyeRose** [comment]
<p>Sorry, didn’t realized that it wasn’t modified from that.</p>
<p>Alright, I reviewed the <a href="https://github.com/benbalter/copy-issue-link-bookmarklet/blob/master/src/bookmark.coffee" rel="nofollow noopener">source code from this script</a>. It looks like the line where we get the title is here:</p>
<pre><code>title = document.getElementsByTagName('h1')[1].children[0].textContent.trim()
</code></pre>
<p>I used the <a href="https://developers.google.com/web/tools/chrome-devtools/console/" rel="nofollow noopener">Google Chrome Developer Tools</a> to try out variations of this line to see what resulted. The <code>.children[0]</code> is particularly interesting to us as it tells us to select the first span tag inside the H1 and to use that as the title. If we used <code>.children[1]</code> however, it gives us the issue number.</p>
<p>Depending on how you want this to be formatted, you could remove the <code>.children[X]</code> entirely and the rest of the code will just give you the raw text data in the issue title which will include the issue number, or you could store the issue number in a separate variable to format the link however you’d like.</p>
<p>Hope that helps! Please let me know if you have additional questions. :slight_smile:</p>

**Comment by fgnievinski** [comment]
I've created a separate community discussion, please upvote it if you'd like to have a new syntax `#123+`:
https://github.com/orgs/community/discussions/41837
